### ISSUE TITLE
set() removes symbol keys

### ISSUE THREAD

**Issue by jetpacmonkey** [planttheidea/unchanged#29]
To reproduce:

```js
const symbol = Symbol();
const obj = { [symbol]: 'foo', nest: { hello: 'world' } };
const updated = set('nest.hello', 'there', obj); // { nest: { hello: 'there' } }, symbol key is removed
```

**Comment by planttheidea** [comment]
Indeed ... as does lodash's `assign` method, as does ramda's `clone` method. In all cases, it only copies over string keys.

That said, this is an edge case thats handled by the native `Object.assign`, so no reason we can't handle it, I just want to make sure I'm not sacrificing too much when it comes to performance. I'm playing around with some things, hopefully I'll have something after the holiday.

**Comment by jetpacmonkey** [comment]
Fair enough. It was a change in behavior that caused an issue when I
upgraded from an earlier version so I wanted to make sure it was called
out.

**Comment by planttheidea** [comment]
You are spot on! I forgot that I used to use the native `Object.assign` for this library, but once I got complains about the need to polyfill to use I switched to using a custom implementation. That makes this a proper regression.

I think I've found a nice compromise, a fallback custom implementation that respects symbols while leveraging the native `Object.assign` when possible. This will also not just apply to set, but all methods (`merge`, `remove`, etc.). I'm just adding test cases around those other methods, should have a PR up soonish.

**Comment by planttheidea** [comment]
Created the PR, feel free to peruse ... I won't merge it until after the holiday, so that I can look at it again with a fresh set of eyes, but I think it satisfies expectations:

- `add`, `merge`, `remove`, `set`, and `transform` will all retain `Symbol`-based properties on objects when cloning
- the native `Object.assign()` is used when available, else it falls back to a custom implementation that handles string and `Symbol` keys

**Comment by planttheidea** [comment]
Alright, fix has been released as `1.5.1`, let me know if you have any more issues!
